pkg/operator/ceph/disruption/clusterdisruption/osd.go

```diff
@@ -129,23 +170,14 @@ func (r *ReconcileClusterDisruption) initializePDBState(request reconcile.Reques
 	err := r.client.Get(context.TODO(), pdbStateMapRequest, pdbStateMap)
 
 	if kerrors.IsNotFound(err) {
-		// create configmap and PDBs for all nodes labeled by failuredomain
-		logger.Infof("inititalizing pod disruption budgets for osds")
-		// one pdb is created per OSD, but after initialization they are created/deleted in failuredomain groups
-		for _, osdData := range osdDataList {
-			err := r.createPDBForOSD(osdData.Deployment)
-			if err != nil {
-				return pdbStateMap, errors.Wrapf(err, "failed to create pdb for osd deployment %q. %v", osdData.Deployment.ObjectMeta.GetName(), err)
-			}
-		}
-		pdbStateMap.Data = map[string]string{disabledPDBKey: ""}
-		// create configmap
+		// create configmap to track the draining failure domain
+		pdbStateMap.Data = map[string]string{drainingFailureDomain: ""}
```

Comment:
Why do we need to track the draining state in a configmap? I would expect we could always compute it based on the state of the OSDs. For example:
- If all OSDs are up and PGs clean, the maxUnavailable=1 PDB is created and other PDBs are deleted
- If an OSD is down or PGs are not clean, create the maxUnavailable=0 PDB for other failure domains, and remove the overall PDB

That may be oversimplified, but I want to clear why we need to store this state.

Reply:
Tracking draining state in configMap helps us in following scenarios:
- A previously drained node comes back up,  **all OSDs are up but PGs are not clean** (cluster is healing) and another node in a different failure domain is drained, we would want to block this new drain. 
- We also track the time when the drain was started in the configmap. This helps us to add maintenance timeout (adding no-out flag to drained OSD for a specified time)
- By tracking time in the configmap, we can also add timeout for cluster healing (that is all pgs active+clean). This could help us solve point 2 mentioned in the [design](https://bugzilla.redhat.com/show_bug.cgi?id=1861104#c43)